src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -433,5 +449,52 @@ void WiFiManager::PostConnectivityStatusChange(ConnectivityChange changeType)
     PlatformMgr().PostEventOrDie(&networkEvent);
 }
 
+System::Clock::Milliseconds32 WiFiManager::GetNextRecoveryTime()
+{
+
+    if (mConnectionRecoveryTimeMs >= kConnectionRecoveryMaxIntervalMs)
+    {
+        // reset previous connection recovery time to max value
+        mConnectionRecoveryTimeMs = kConnectionRecoveryMaxIntervalMs;
+        // find the new random jitter value in range [-jitter, +jitter]
+        int32_t jitter            = chip::Crypto::GetRandU32() % (2 * jitter + 1) - jitter;
+        mConnectionRecoveryTimeMs = mConnectionRecoveryTimeMs + jitter;
+    }
+    else
+    {
+        mConnectionRecoveryTimeMs = mConnectionRecoveryTimeMs * 2;
+    }
+    return System::Clock::Milliseconds32(mConnectionRecoveryTimeMs);
+}
+
+void WiFiManager::Recover(System::Layer * layer, void * param)
```

Comment:
We can skip arguments' names, some compilers may complain that these are unused.